### PR TITLE
Range-ify std.file rmdir()

### PR DESCRIPTION
**Pull request by WalterBright**
analogous to changes made to chdir()

### CODE REVIEW

**Comment by schuetzm** [comment]
Why don't you pull the duplication out of the version blocks? As long as we have only Windows and Posix and no other platform that requires something other than C strings...

File: std/file.d [hunk 1]
```diff
@@ -1864,19 +1864,40 @@ unittest
...
+
     version(Windows)
     {
-        cenforce(RemoveDirectoryW(pathname.tempCStringW()),
-                pathname);
+        static auto trustedRemoveDirectoryW(const(FSChar)* pathz) @trusted
+        {
+            return RemoveDirectoryW(pathz);
+        }
+        static if (isNarrowString!R && is(Unqual!(ElementEncodingType!R) == char))
```
Comment (DmitryOlshansky) [inline std/file.d]: Indeed no need to duplicate across version blocks.

**Comment by DmitryOlshansky** [comment]
@WalterBright suggest you to ping reviewers when something changes - there is no notificaition that pull #xyz changed otherwise.

**Comment by WalterBright** [comment]
Consider yourself pinged, then!

**Comment by DmitryOlshansky** [comment]
> Consider yourself pinged, then!

Ironically I got it and then forgot about it :)

**Comment by DmitryOlshansky** [comment]
Auto-merge toggled on

**Pull request merged**